app/controllers/topics_controller.rb

```diff
@@ -803,8 +803,19 @@ def invite
   end
 
   def set_notifications
+    user =
+      if is_api? && @guardian.is_admin?
+        begin
+          fetch_user_from_params
+        rescue Discourse::NotFound
```

Comment:
Hmm, yeah, `fetch_user_from_params` returns `Discourse::NotFound` if the user isn't found.... not ideal for our use case here. A blanket `rescue` isn't great though, it can hide other issues. 

Maybe we can explicitly check for `params[:username]` or `params[:external_id]` and only in those cases use `fetch_user_from_params`?

Reply:
Just wondering, what if an invalid `username` or `external_id` is passed? We'd still need to `rescue` right?

Reply:
```ruby
    user = User.find_by_username(params[:username]) if is_api? && @guardian.is_admin?
    user ||= current_user
```

Reply:
> Just wondering, what if an invalid `username` or `external_id` is passed? We'd still need to `rescue` right?

`fetch_user_from_params` already handles that correctly. It's best to use it here so we get consistent API behaviour.